jkube-kit/build/service/docker/src/test/java/org/eclipse/jkube/kit/build/service/docker/auth/EnvironmentVariablesTestUtil.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2019 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at:
+ *
+ *     https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.jkube.kit.build.service.docker.auth;
+
+import java.lang.reflect.Field;
+import java.util.Collections;
+import java.util.Map;
+
+/**
+ * Utility class for modifying environment variables in tests.
+ * Uses reflection to modify the environment map.
+ */
+public class EnvironmentVariablesTestUtil {
```

Comment:
This is kind of fragile.

We have other tests that depend on the environment.
Maybe it's a good time to implement a proper Environment wrapper (in a separate PR) that can be leveraged by this feature's production code so that it can be set in tests.